src/stan/services/optimize/defaults.hpp

```diff
@@ -0,0 +1,326 @@
+        }
+
+        /**
+         * Validates tol_grad; tol_grad must be greater than or equal to 0.
+         *
+         * @param[in] tol_grad argument to validate
+         * @throw std::invalid argument if tol_grad < 0
+         */
+        static void validate(double tol_grad) {
+          if (!(tol_grad >= 0))
+            throw std::invalid_argument("tol_grad must be greater "
+                                        "than or equal to 0");
+        }
+
+        /**
+         * Default tol_grad: 1e-8.
+         *
+         * @returns default
+         */
+        static double default_value() {
+          return 1e-8;
+        }
+      };
+
+
+      /**
+       * Convergence tolerance on the relative norm of the gradient.
+       */
+      struct tol_rel_grad {
+        /**
+         * Description of tol_rel_grad.
+         *
+         * @returns description
+         */
+        static std::string description() {
+          return "Convergence tolerance on the relative norm of the gradient.";
+        }
+
+        /**
+         * Validates tol_rel_grad; tol_rel_grad must be greater than
+         * or equal to 0.
+         *
+         * @param[in] tol_rel_grad argument to validate
+         * @throw std::invalid argument if tol_rel_grad < 0
+         */
+        static void validate(double tol_rel_grad) {
+          if (!(tol_rel_grad >= 0))
+            throw std::invalid_argument("tol_rel_grad must be greater "
+                                        "than or equal to 0.");
+        }
+
+        /**
+         * Default tol_rel_grad: 10000000
+         *
+         * @returns default
+         */
+        static double default_value() {
+          return 10000000;
+        }
+      };
+
+      /**
+       * Convergence tolerance on changes in parameter value.
+       */
+      struct tol_param {
+        /**
+         * Description of tol_param.
+         *
+         * @returns description
+         */
+        static std::string description() {
+          return "Convergence tolerance on changes in parameter value.";
+        }
+
+        /**
+         * Validates tol_param; tol_param must be greater than or equal to 0.
+         *
+         * @param[in] tol_param argument to validate
+         * @throw std::invalid argument if tol_param < 0
+         */
+        static void validate(double tol_param) {
+          if (!(tol_param >= 0))
+            throw std::invalid_argument("tol_param");
+        }
+
+        /**
+         * Default tol_param: 1e-08.
+         *
+         * @returns default
+         */
+        static double default_value() {
+          return 1e-08;
+        }
+      };
+
+      /**
+       * Amount of history to keep for L-BFGS.
+       */
+      struct history_size {
+        /**
+         * Description of history_size.
+         *
+         * @returns description
+         */
+        static std::string description() {
+          return "Amount of history to keep for L-BFGS.";
```

Comment:
I take it then you'll say the same thing about all the rewording I was suggesting.  This is going to be a long series of pull requests.  I think it may be easier just to review everything again because nobody's creating these issues when you say things are out of scope.  I don't really want to create an issue to change a single word in an output.  We haven't really nailed down the API at that level.  Is it problematic because of all the brittle exact-wording tests would require major rewriting of tests?